integrationtests/tests/tests.rs

```diff
@@ -89,9 +89,10 @@ async fn minted_coins_cannot_double_spent_with_different_nodes() {
     let out3 = user3.client.reissue(coins, rng()).await.unwrap();
     fed.run_consensus_epochs(2).await; // process transaction + sign new coins
 
-    // FIXME is this the correct behavior, that the first one goes through?
-    assert!(user2.client.fetch_coins(out2).await.is_ok());
-    assert!(user3.client.fetch_coins(out3).await.is_err());
+    //It's common knowledge among users (clients respectively) that just receiving 'ecash' does not imply being paid successfully.
+    //Clients will only accept a payment that can be successfully fetched, so this is not a UX concern.
+    assert!(user2.client.fetch_coins(out2).await.is_ok()); //user2 received the payment
+    assert!(user3.client.fetch_coins(out3).await.is_err()); //user3 did not
```

Comment:
I think this has more to do with our conflict filtering. The first one (in consensus order, so maybe not deterministic) goes through, all others fail. 

In this case it would be more correct to just count the successful spends and they should be 1. It would also be interesting to test with a different setup where there is an epoch between the first and the second spend since that's a  different code path.

The thing you mention, that clients can already accept payments once the tx is confirmed even if they can't fetch e-cash tokens yet, isn't implemented yet and a different issue imo.

Reply:
I'd say `user2.(..).is_ok()` AND `user3..n.(..).is_err()` <=> "only one spend", I wouldn't know how to convey that more correct in the code (In theory we could create a user3..n lazy generator and ask if it's user.next is err until we are satisfied but that would be wired xd). Should I just remove the comments or does this test need to be altered ?

Reply:
now that I read it again, I see my missunderstanding: we don't really know which user will be paid but exactly one will be paid..(so forget my comment above). Maybe we could wait for the status to be valid before reissueing `user3` ?

Reply:
I could add [this](https://github.com/fedimint/minimint/pull/55/files#diff-8ed9f210e336dbfc7d15e9201382a09ce4f08de95e060d35d949674b0c74e1c5R389-R416) to `user.rs` and edit `tests.rs` to wait for the accepted tx ?

Reply:
We could just xor the payment status of the both users. But probably best to ask @jkitman, he has written the test initially.

Reply:
Yeah I wouldn't overengineer the assertion, we just need to ensure no double spend took place regardless of who got paid.  Something like:
```
assert_eq!(user2.total_coins() + user3.total_coins(), sats(2000));
assert!(user2_response.is_err() || user3_response.is_err());
```

Reply:
pushed again with your suggestion @jkitman